### PR TITLE
(MODULES-6454) - PDK Convert mysql

### PR DESCRIPTION
**Pull request by pmcmaw**
Using PDK to convert module so that it is compliant with the PDK.

The second commit addresses a bundler issue MODULES-6339.
The next time PDK convert is ran it will re-add the newer version of the bundler gem.

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -1,8 +1,7 @@
 ---
-require:
- - rubocop-rspec
- - rubocop-i18n
```
Comment (pmcmaw) [inline .rubocop.yml]: @HelenCampbell we may need this.

**Comment by pmcmaw** [comment]
`GetText/DecorateString` is disabled as this rule flags that `desc` and `@doc`. Currently this is not something we need to decorate. 

I have tested this PR to ensure failures are caught in Travis when necessary ruby and puppet strings are not decorated.

Ruby:
```
bundle exec rake rubocop
lib/puppet/provider/mysql.rb:63:28: C: GetText/DecorateFunctionMessage: 'raise' function, message should be decorated. 
      raise Puppet::Error, "#mysql_caller: Unrecognised type '%{type}'" % { type: type }
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
Puppet:
```
bundle exec rake release_checks
manifests/server.pp - WARNING: 'warning' messages should be decorated: eg translate(The `old_root_password` attribute is no longer used and will be removed in a future release.) on line 48
```

File: .sync.yml [hunk 2]
```diff
@@ -1,13 +1,69 @@
...
+  spec_overrides:
+    -  "require 'spec_helper_local'"
+
 .rubocop.yml:
-  unmanaged: true
-.rubocop_todo.yml:
-  unmanaged: true
+   default_configs:
+    inherit_from: .rubocop_todo.yml
+    require:
```
Comment (pmcmaw) [inline .sync.yml]: By adding the `rubocop-rspec` require here it allows us to require multiple things. If I only require `rubocop-i18n` the `rubocop-rspec` from the template it overwritten.

Comment (pmcmaw) [reply .sync.yml]: `rubocop-i18n` is required because this module has been internationalized.

File: .sync.yml [hunk 3]
```diff
@@ -1,13 +1,69 @@
...
+spec/spec_helper.rb:
+  spec_overrides:
+    -  "require 'spec_helper_local'"
+
 .rubocop.yml:
-  unmanaged: true
-.rubocop_todo.yml:
-  unmanaged: true
+   default_configs:
+    inherit_from: .rubocop_todo.yml
```
Comment (pmcmaw) [inline .sync.yml]: The `.rubocop_todo.yml` is required as it contains configuration to disable a cop. The cop is currently not needed.

File: .sync.yml [hunk 4]
```diff
@@ -1,13 +1,69 @@
...
+        from_env: BEAKER_RSPEC_VERSION
+    ':development':
+      - gem: puppet-blacksmith
+        version: '~> 3.4'
+      - gem: puppet-lint-i18n
+
+Rakefile:
+  requires:
+    - puppet_blacksmith/rake_tasks
+    - puppet_pot_generator/rake_tasks
```
Comment (pmcmaw) [inline .sync.yml]: This is required for i18n.

**Comment by pmcmaw** [comment]
@HelenCampbell I have updated this PR with all the necessary changes to ensure that it has all the i18n tools and that they work. I have commented on changes that differ compared to the usual PDK convert PRs. It would be great if I could get another review now this has all been done.

**Comment by pmcmaw** [comment]
This has ran through the adhoc pipeline with no issues.

File: Rakefile [hunk 5]
```diff
@@ -1,37 +1,4 @@
 require 'puppetlabs_spec_helper/rake_tasks'
-require 'puppet-lint/tasks/puppet-lint'
```
Comment (HelenCampbell) [inline Rakefile]: This needs added to .sync.yml

**Pull request merged**